pkg/metrics/metrics.go

```diff
@@ -34,73 +65,73 @@ var (
 	// controller, or the Prometheus exporter will error. b/247516388
 	// https://github.com/GoogleContainerTools/kpt-resource-group/blob/main/controllers/metrics/metrics.go#L88
 	PipelineError = stats.Int64(
-		"pipeline_error_observed",
+		PipelineErrorName,
 		"A boolean value indicates if error happened at readiness stage when syncing a commit",
 		stats.UnitDimensionless)
 
 	// ReconcileDuration metric measures the latency of reconcile events.
 	ReconcileDuration = stats.Float64(
-		"reconcile_duration_seconds",
+		ReconcileDurationName,
 		"The duration of reconcile events in seconds",
 		stats.UnitSeconds)
 
 	// ParserDuration metric measures the latency of the parse-apply-watch loop.
 	ParserDuration = stats.Float64(
-		"parser_duration_seconds",
+		ParserDurationName,
 		"The duration of the parse-apply-watch loop in seconds",
 		stats.UnitSeconds)
 
 	// LastSync metric measures the timestamp of the latest Git sync.
 	LastSync = stats.Int64(
-		"last_sync_timestamp",
+		LastSyncName,
 		"The timestamp of the most recent sync from Git",
 		stats.UnitDimensionless)
 
 	// DeclaredResources metric measures the number of declared resources parsed from Git.
 	DeclaredResources = stats.Int64(
-		"declared_resources",
+		DeclaredResourcesName,
 		"The number of declared resources parsed from Git",
 		stats.UnitDimensionless)
 
 	// ApplyOperations metric measures the number of applier apply events.
 	ApplyOperations = stats.Int64(
-		"apply_operations",
+		ApplyOperationsName,
```

Comment:
This looks like it changes the name to include the "_total" suffix. Was that intentional? Same with the other totals

Reply:
If you look at view.go it was appending the `_total` suffix before, and that's actually where the metric names are finalized when exporting. The const-fication puts them in one place to rule out the confusion. I've checked both Kmetrics and resource group controller, they two don't have the name discrepancy between metric.go and view.go. Agreed that pulling this out to a separate PR may be better, will cleanup a bit along with the declared_resources chagne.